SAP/spartacus
# fix: context.error was not printed because JSON.stringify(Error) doesn't work by default in JS

**Changes:**
- While stringifying the logged JSON, now we're converting each property being an Error instance into a pretty string. Otherwise, by default JSON.stringify would output by default only `{}`.
- Increased the `depth` option for `formatWithOptions()` util.  Otherwise, the Errors's properties at nested levels higher than 2 (e.g. `error.cause.headers`, where `cause` is `HttpErrorResponse`) were not be visible (e.g. abbreviated as `Map`) in the logs.

**QA steps:**
1. in app.module.ts provide the following config chunk (causing HTTP call to backend to fail):
   ```ts
    provideConfig({
      backend: {
        occ: {
          endpoints: {
            components: 'some-fake-endpoint',
          },
        },
      },
    }),
   ```
2. start SSR server - run `npm run dev:ssr` 
3. in another terminal `curl http://localhost:4200`
4. In the logs of the SSR server, see that under the message `"message": "Request is resolved with the SSR rendering error (/)` there is an property "error" that is not `{}`, but an actual string with Error details

BEFORE:
![image](https://github.com/user-attachments/assets/1b9d2115-696c-4d9d-82cc-b372ff94fff3)


AFTER:
![image](https://github.com/user-attachments/assets/ab8f02a6-5f26-4605-8f96-5a7d8ad91f11)

Future outlook: in another PR we plan to improve the readability of the error messages by displaying them in multi-line (instead of a single line with `\\n`).

fixes https://jira.tools.sap/browse/CXSPA-8360
